src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.cs

```diff
@@ -134,12 +134,31 @@ protected void SkipToAndParseCode(Func<HtmlSymbol, bool> condition)
                 {
                     if (last != null)
                     {
-                        Accept(last);
+                        if (!startOfLine || last.Type != HtmlSymbolType.WhiteSpace)
+                        {
+                            Accept(last);
+                        }
+
                         last = null;
                     }
+
                     AddMarkerSymbolIfNecessary();
                     Output(SpanKind.Markup);
+
                     RazorComment();
+
+                    // Handle the whitespace and newline at the end of a razor comment.
+                    if (startOfLine &&
```

Comment:
Just checked. The razor comments in the other two places are working fine as is. No need to add this logic there.